hibernate-core/src/main/java/org/hibernate/persister/entity/EntityPersister.java

```diff
@@ -373,15 +373,23 @@ Serializable loadEntityIdByNaturalId(
 	/**
 	 * Load an instance of the persistent class.
 	 */
-	Object load(Serializable id, Object optionalObject, LockMode lockMode, SharedSessionContractImplementor session)
+	Object load(Serializable id, Object optionalObject, LockMode lockMode, SharedSessionContractImplementor session, Boolean readOnly)
 	throws HibernateException;
 
+	default Object load(Serializable id, Object optionalObject, LockMode lockMode, SharedSessionContractImplementor session) {
+		return load( id, optionalObject, lockMode, session, null );
+	}
+
 	/**
 	 * Load an instance of the persistent class.
 	 */
-	Object load(Serializable id, Object optionalObject, LockOptions lockOptions, SharedSessionContractImplementor session)
+	Object load(Serializable id, Object optionalObject, LockOptions lockOptions, SharedSessionContractImplementor session, Boolean readOnly)
 	throws HibernateException;
 
+	default Object load(Serializable id, Object optionalObject, LockOptions lockOptions, SharedSessionContractImplementor session) {
```

Comment:
The default implementation should actually be the one with the new signature in order to remain compatibility with custom persisters.

Reply:
Well, in this case the new method generalizes the old one for it accepts a new parameter while the old one doesn't. For that reason, it seems hiding the old one (it is still overridable for sure) and exposing the new one is better. Maybe I need to add `@Deprecated` to the old method to make it explicit.

Reply:
Exactly! Thanks.